tarn/location/interface.py

```diff
@@ -41,15 +41,11 @@ def read_batch(self, keys: Keys) -> Iterable[Tuple[Key, Union[None, Tuple[Value,
     def contents(self) -> Iterable[Tuple[Key, Self, Meta]]:
         pass
 
-
-class Writable(Location, ABC):
-    @abstractmethod
     def write(self, key: Key, value: Value, labels: MaybeLabels) -> ContextManager[MaybeValue]:
         pass
 
-    @abstractmethod
     def delete(self, key: Key) -> bool:
-        pass
+        return False
```

Comment:
make these methods abstract, and create a simple `ReadOnly` class that has this "no-read" behaviour